packages/gtmapi/lmsrvcore/api/mutations/chunkupload.py

```diff
@@ -92,13 +75,38 @@ def validate_args(args):
 
     @staticmethod
     def py_secure_filename(filename: str) -> str:
-        # Take any non-alphanumeric (and some limited special chars) and replace with _
-        # Also note, max filename size is 4096 chars
-        s = lambda n: n if (n.isalnum() or n in '._-+=') else '_'
-        safe_fname = ''.join([s(c) for c in os.path.basename(filename)])[:255]
-        if safe_fname != filename:
-            logger.info(f"Renaming unsafe filename `{filename}` to `{safe_fname}`")
-        return safe_fname
+        """Method to clean up provided filenames to be safe, relative paths
+
+        This function removes leading slashes, control characters, '..' and '.' in directories, and replaces \/:*"<>|?
+        with underscores.
+
+
+        Args:
+            filename: Filename to sanitize
+
+        Returns:
+            str
+        """
+        # Completely remove control characters
+        safe_filename = "".join(c for c in filename if unicodedata.category(c)[0] != "C")
+
+        # Remove leading slash if attempting an absolute path
+        if safe_filename[0] == os.path.sep:
+            safe_filename = safe_filename[1:]
+
+        # Remove ../ or ./ paths (should only be relative names from the repo root)
+        filename_parts = Path(safe_filename).parts
+        relative_filename_parts = [p for p in filename_parts if p not in ['..', '.']]
+
+        # Replace invalid characters with underscores
+        invalid_char_map = {ord(ch): '_' for ch in '\\/:*"<>|?'}
+        safe_parts = [part.translate(invalid_char_map).strip() for part in relative_filename_parts]
+
+        safe_filename = os.path.sep.join(safe_parts)
```

Comment:
If you're already using pathlib...

```suggestion
        safe_filename = str(Path(safe_parts))
```

Reply:
this doesn't work...but i made it a bit more explicit.